x/emissions/keeper/msgserver/msg_server_worker_payload_test.go

```diff
@@ -159,7 +165,11 @@ func (s *MsgServerTestSuite) TestMsgInsertWorkerPayloadNotFailsWithNilInference(
 	blockHeight := workerMsg.WorkerDataBundle.InferenceForecastsBundle.Forecast.BlockHeight
 	ctx = ctx.WithBlockHeight(blockHeight)
 
-	_, err := msgServer.InsertWorkerPayload(ctx, &workerMsg)
+	// Add worker to topic whitelist
+	err := s.emissionsKeeper.AddToTopicWorkerWhitelist(ctx, workerMsg.WorkerDataBundle.TopicId, workerMsg.WorkerDataBundle.Worker)
```

Comment:
Do we have "negative" tests? 
Could be done just like 
```
	_, err := msgServer.InsertWorkerPayload(ctx, &workerMsg)
	require.Error(err, "InsertWorkerPayload should return an error before adding worker to whitelist")

	// Add worker to topic whitelist
	err = s.emissionsKeeper.AddToTopicWorkerWhitelist(ctx, workerMsg.WorkerDataBundle.TopicId, workerMsg.WorkerDataBundle.Worker)
	require.NoError(err, "InsertWorkerPayload should not return an error after adding worker to whitelist")
```

Reply:
great call. just added